### PR TITLE
[stable] Lower lint level for READ_ZERO_BYTE_VEC

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
This avoids reporting false-positives; see rust-lang/rust-clippy#9274 for details.

cc @rust-lang/clippy -- do we want a direct change landed on stable here? If so, please r+ this PR, otherwise we can just close it. Would appreciate confirmation this is the right change to make as well.

cc @joshtriplett -- filing due to https://rust-lang.zulipchat.com/#narrow/stream/301329-t-devtools/topic/clippy.20false.20positive

### CODE REVIEW

File: src/tools/clippy/clippy_lints/src/read_zero_byte_vec.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ declare_clippy_lint! {
     /// ```
     #[clippy::version = "1.63.0"]
     pub READ_ZERO_BYTE_VEC,
-    correctness,
+    pedantic,
```
Comment (Manishearth) [inline src/tools/clippy/clippy_lints/src/read_zero_byte_vec.rs:46]:
```suggestion
    nursery,
```

**Review by Manishearth — APPROVED** [review]

**Comment by flip1995** [comment]
You will probably have to run `cd src/tools/clippy && cargo dev update_lints`, so that the lint gets moved out of the default group.

Otherwise this sgtm

File: src/tools/clippy/clippy_lints/src/read_zero_byte_vec.rs (hunk 1 above)
Comment (Manishearth) [reply src/tools/clippy/clippy_lints/src/read_zero_byte_vec.rs:46]: Also this isn't enough, you need to run `cargo dev update_lints`

**Review by Manishearth — CHANGES_REQUESTED** [review]
r=me 

but you need to run `cargo dev update_lints` to update the lint categories

File: src/tools/clippy/clippy_lints/src/lib.register_nursery.rs [hunk 2]
```diff
@@ -25,11 +25,14 @@ store.register_group(true, "clippy::nursery", Some("clippy_nursery"), vec![
...
     LintId::of(nonstandard_macro_braces::NONSTANDARD_MACRO_BRACES),
     LintId::of(option_if_let_else::OPTION_IF_LET_ELSE),
+    LintId::of(read_zero_byte_vec::READ_ZERO_BYTE_VEC),
     LintId::of(redundant_pub_crate::REDUNDANT_PUB_CRATE),
     LintId::of(regex::TRIVIAL_REGEX),
     LintId::of(strings::STRING_LIT_AS_BYTES),
     LintId::of(suspicious_operation_groupings::SUSPICIOUS_OPERATION_GROUPINGS),
     LintId::of(trailing_empty_array::TRAILING_EMPTY_ARRAY),
+    LintId::of(trait_bounds::TRAIT_DUPLICATION_IN_BOUNDS),
+    LintId::of(trait_bounds::TYPE_REPETITION_IN_BOUNDS),
```
Comment (Mark-Simulacrum) [inline src/tools/clippy/clippy_lints/src/lib.register_nursery.rs:35]: This seems right according to the content of the lint definition but obviously isn't a change I made in this PR -- is this OK?

**Review by flip1995 — APPROVED** [review]
LGTM. the other changes are from a limbo state this lint was in. Probably due to a weird sync.

Those files are removed anyway with the next sync, so this change is fine :+1:

**Pull request merged**